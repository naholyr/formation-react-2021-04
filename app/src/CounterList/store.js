import { combineReducers, createStore } from "redux";

export const initialState = {
  counters: {
    nbCounters: 0,
  },
};

export const countersReducer = (state = initialState.counters, action) => {
  switch (action.type) {
    case "CHANGE_NB_COUNTERS":
      return {
        ...state,
        nbCounters: action.payload.nb,
      };
    default:
      return state;
  }
};

export const reducer = combineReducers({
  counters: countersReducer,
});

export const initStore = () => {
  return createStore(reducer);
};
