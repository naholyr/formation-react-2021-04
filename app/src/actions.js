export const setUser = (name) => {
  return {
    type: "SET_USER",
    payload: { name },
  };
};

export const unsetUser = () => ({
  type: "SET_USER",
  payload: { name: null },
});
