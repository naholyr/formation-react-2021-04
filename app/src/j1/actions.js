// Action creator
export const changeNbCounters = (nb) => {
  return {
    type: "CHANGE_NB_COUNTERS",
    payload: { nb },
  };
};

export const resetNbCounters = () => {
  return {
    type: "CHANGE_NB_COUNTERS",
    payload: { nb: 0 },
  };
};
