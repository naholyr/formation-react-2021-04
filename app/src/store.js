import { combineReducers, createStore } from "redux";

export const initialState = {
  user: {
    name: null,
  },
  game: {
    trials: [
      {
        name: "Coco",
        word: [
          ["B", 0],
          ["O", 0],
          ["U", 1],
          ["L", 0],
          ["I", 1],
          ["M", 1],
          ["I", 1],
          ["Q", 0],
          ["U", 1],
          ["E", 1],
        ],
        win: false,
      },
      {
        name: "Nicolas",
        word: [
          ["B", 0],
          ["I", 1],
          ["J", 0],
          ["O", 0],
          ["U", 1],
          ["T", 1],
          ["I", 1],
          ["E", 2],
          ["R", 1],
          ["S", 1],
        ],
        win: false,
      },
      {
        name: "Coco",
        word: [
          ["T", 1],
          ["E", 1],
          ["L", 0],
          ["E", 1],
          ["V", 0],
          ["I", 1],
          ["S", 1],
          ["I", 1],
          ["O", 0],
          ["N", 1],
        ],
        win: false,
      },
      {
        name: "Nicolas",
        word: [
          ["A", 0],
          ["B", 0],
          ["R", 1],
          ["I", 1],
          ["C", 0],
          ["O", 0],
          ["T", 1],
          ["I", 1],
          ["E", 1],
          ["R", 1],
        ],
        win: false,
      },
    ],
    scores: [
      {
        name: "Renaud",
        score: 9,
      },
      {
        name: "test",
        score: 7,
      },
      {
        name: "Coco",
        score: 0,
      },
      {
        name: "Romain F",
        score: 0,
      },
      {
        name: "ben",
        score: 0,
      },
      {
        name: "Nicolas",
        score: 0,
      },
    ],
    wordLength: 10,
    inputDisabled: false,
  },
};

export const userReducer = (state = initialState.user, action) => {
  switch (action.type) {
    case "SET_USER":
      return { ...state, name: action.payload.name };
    default:
      return state;
  }
};

export const gameReducer = (state = initialState.game, action) => {
  switch (action.type) {
    default:
      return state;
  }
};

export const reducer = combineReducers({
  user: userReducer,
  game: gameReducer,
});

export const initStore = () => {
  return createStore(
    reducer,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  );
};
