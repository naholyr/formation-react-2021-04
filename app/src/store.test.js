import { setInputDisabled, setUser, startNewGame } from "./actions";
import { reducer, initialState } from "./store";

describe("Store", () => {
  describe("Reducer", () => {
    it("should init state", () => {
      const prevState = undefined;
      const action = { type: "@@INIT" };
      const newState = reducer(prevState, action);
      expect(newState).toEqual(initialState);
    });

    it("should set username", () => {
      const prevState = undefined;
      const action = setUser("toto");
      const newState = reducer(prevState, action);
      expect(newState).toMatchSnapshot();
    });

    it("should disable input", () => {
      const prevState = {
        ...initialState,
        game: { ...initialState.game, inputDisabled: true },
      };
      const action = setInputDisabled(false);
      const newState = reducer(prevState, action);
      expect(newState).toMatchObject({ game: { inputDisabled: false } });
    });

    it("should start new game", () => {
      const prevState = {
        ...initialState,
        game: {
          ...initialState.game,
          trials: ["FAKE_TRIAL1", "FAKE_TRIAL2"],
        },
      };
      const action = startNewGame(7);
      const newState = reducer(prevState, action);
      expect(newState).toMatchObject({ game: { wordLength: 7, trials: [] } });
    });
  });

  describe("Selectors", () => {
    it.todo("should select username");
  });
});
