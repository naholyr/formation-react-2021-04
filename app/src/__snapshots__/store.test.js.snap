// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store Reducer should set username 1`] = `
Object {
  "game": Object {
    "inputDisabled": true,
    "loaded": false,
    "scores": Array [],
    "trials": Array [],
    "wordLength": 0,
  },
  "user": Object {
    "name": "toto",
  },
}
`;
